spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketSession.java

```diff
@@ -51,52 +70,100 @@ public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFact
 	public JettyWebSocketSession(Session session, HandshakeInfo info, DataBufferFactory factory,
 			@Nullable Sinks.Empty<Void> completionSink) {
 
-		super(session, ObjectUtils.getIdentityHexString(session), info, factory, completionSink);
-		// TODO: suspend causes failures if invoked at this stage
-		// suspendReceiving();
+		super(session, ObjectUtils.getIdentityHexString(session), info, factory);
+		this.handlerCompletionSink = completionSink;
+		this.flux = Flux.create(emitter -> {
+			this.sink = emitter;
+			emitter.onRequest(n -> {
+				boolean demand = false;
+				this.lock.lock();
+				try {
+					this.requested += n;
+					if (!this.awaitingDemand && this.requested > 0) {
+						this.requested--;
+						this.awaitingDemand = true;
```

Comment:
Should this be `this.awaitingDemand = this.requested == 0`?

If `n > 1` does this still work?

Reply:
Yes this should still work.

We set `awaitingDemand` to true, then call `demand()`. 
So now we're awaiting a corresponding call to `handleMessage`.
If we were not `awaitingDemand` in `handleMessage` it throws ISE, otherwise we reset it to false.

Then in `handleMessage` we repeat, if there is still `requested > 0` we set `awaitingDemand` true and demand again.

Reply:
Can you rename `awaitingDemand` to something better such as `stalled` or `awaitingMessage` or `messageAvailable`?

Reply:
done